src/server/game/Instances/InstanceScript.cpp

```diff
@@ -228,17 +228,20 @@ void InstanceScript::UpdateDoorState(GameObject* door)
     for (; range.first != range.second && open; ++range.first)
     {
         DoorInfo const& info = range.first->second;
-        switch (info.type)
+        switch (info.Behavior)
         {
-            case DOOR_TYPE_ROOM:
+            case EncounterDoorBehavior::OpenWhenNotInProgress:
                 open = (info.bossInfo->state != IN_PROGRESS);
                 break;
-            case DOOR_TYPE_PASSAGE:
+            case EncounterDoorBehavior::OpenWhenDone:
                 open = (info.bossInfo->state == DONE);
                 break;
-            case DOOR_TYPE_SPAWN_HOLE:
+            case EncounterDoorBehavior::OpenWhenInProgress:
                 open = (info.bossInfo->state == IN_PROGRESS);
                 break;
+            case EncounterDoorBehavior::OpenWhenNotDone:
```

Comment:
Out of curiosity, where did this type come from?

Reply:
for anduin bridge after encounter end. Bridges work opposite way, open = you cannot pass, closed = you can pass

Reply:
They really like entangling stuff.